nighthawk/source/client/output_formatter_impl.cc

```diff
@@ -0,0 +1,79 @@
+#include "nighthawk/source/client/output_formatter_impl.h"
+
+#include <chrono>
+#include <sstream>
+
+#include "common/protobuf/utility.h"
+
+namespace Nighthawk {
+namespace Client {
+
+OutputFormatterImpl::OutputFormatterImpl(Envoy::TimeSource& time_source, const Options& options,
+                                         const std::vector<StatisticPtr>& merged_statistics,
+                                         const std::map<std::string, uint64_t>& merged_counters)
+    : time_source_(time_source), options_(options), merged_statistics_(merged_statistics),
+      merged_counters_(merged_counters) {}
+
+ConsoleOutputFormatterImpl::ConsoleOutputFormatterImpl(
+    Envoy::TimeSource& time_source, const Options& options,
+    const std::vector<StatisticPtr>& merged_statistics,
+    const std::map<std::string, uint64_t>& merged_counters)
+    : OutputFormatterImpl(time_source, options, merged_statistics, merged_counters) {}
+
+std::string ConsoleOutputFormatterImpl::toString() const {
+  std::stringstream s;
+
+  // TODO(oschaaf): echo non-default options to CLI output?
+  s << "Merged statistics:\n";
+  for (auto& statistic : merged_statistics_) {
+    if (statistic->count() > 0) {
+      s << fmt::format("{}: {}\n", statistic->id(), statistic->toString());
+    }
+  }
+  s << "\nMerged counters\n";
+  for (auto counter : merged_counters_) {
+    s << fmt::format("counter {}:{}\n", counter.first, counter.second);
+  }
+  return s.str();
+}
+
+Envoy::ProtobufTypes::MessagePtr OutputFormatterImpl::toProto() const {
+  nighthawk::client::Output output;
+  output.set_allocated_options(options_.toCommandLineOptions().release());
+  *(output.mutable_timestamp()) = google::protobuf::util::TimeUtil::NanosecondsToTimestamp(
```

Comment:
This looks much nicer. Two things:

1. Can you use the `Envoy::Protobuf` namespace alias when referring to `google::protobuf` (here and elsewhere in NH if there are such places)? We need to do namespace remapping on Google import, and this assists with that. 

2. Where do you get that `time_since_epoch` is nanosecs here? Should we have an explicit duration cast?

Reply:
1. Done
2. Ugh; nice catch, added an explicit duration cast. On my system this resolves to nanoseconds..  but digging into this for a bit, this looks platform specific.